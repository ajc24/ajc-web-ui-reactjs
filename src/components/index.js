/**
 * Developed by Anthony Cox in 2024
 */

/* Export the header components */
export { default as HeaderTypeA } from './header/HeaderTypeA';

/* Export the main components */
export { default as MainContent } from './main/MainContent';

/* Export the menu bar components */
export { default as MenuBarTypeA } from './menu-bar/MenuBarTypeA';
