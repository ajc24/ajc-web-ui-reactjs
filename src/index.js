/**
 * Developed by Anthony Cox in 2024
 */

/* Import the header components */
import Header from './components/header/Header';

/* Import the image components */
import DecorativeImage from './components/images/DecorativeImage';
import FunctionalImage from './components/images/FunctionalImage';
import InformativeImage from './components/images/InformativeImage';

/* Export all components */
export {
  DecorativeImage,
  FunctionalImage,
  InformativeImage,
  Header,
};
