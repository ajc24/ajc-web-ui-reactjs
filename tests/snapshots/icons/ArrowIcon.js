/**
 * Developed by Anthony Cox in 2025
 */
import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-testing-code';
import { ArrowIcon } from '../../../src';

describe('Arrow Icon', () => {
  /* Set the IDs for use in the tests */
  const testComponentId = 'test-component-id';

  describe('Arrow icon with size 1, with black colour, with down direction', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ArrowIcon colour="black" direction="down" id={testComponentId} size="1" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Arrow icon with size 2, with gold colour, with left direction', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ArrowIcon colour="gold" direction="left" id={testComponentId} size="2" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Arrow icon with default size, with white colour, with right direction', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ArrowIcon colour="white" direction="right" id={testComponentId} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Arrow icon with default size, with default colour, with up direction', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ArrowIcon direction="up" id={testComponentId} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
