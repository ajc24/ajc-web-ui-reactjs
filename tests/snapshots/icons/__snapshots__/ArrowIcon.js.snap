// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow Icon Arrow icon with default size, with default colour, with up direction verifies the snapshot for the component 1`] = `"<span id="test-component-id--arrow-icon-up" class="arrow-icon arrow-icon-size-one arrow-icon-border-black arrow-icon-direction-up" aria-hidden="true"></span>"`;

exports[`Arrow Icon Arrow icon with default size, with white colour, with right direction verifies the snapshot for the component 1`] = `"<span id="test-component-id--arrow-icon-right" class="arrow-icon arrow-icon-size-one arrow-icon-border-white arrow-icon-direction-right" aria-hidden="true"></span>"`;

exports[`Arrow Icon Arrow icon with size 1, with black colour, with down direction verifies the snapshot for the component 1`] = `"<span id="test-component-id--arrow-icon-down" class="arrow-icon arrow-icon-size-one arrow-icon-border-black arrow-icon-direction-down" aria-hidden="true"></span>"`;

exports[`Arrow Icon Arrow icon with size 2, with gold colour, with left direction verifies the snapshot for the component 1`] = `"<span id="test-component-id--arrow-icon-left" class="arrow-icon arrow-icon-size-two arrow-icon-border-gold arrow-icon-direction-left" aria-hidden="true"></span>"`;
